i3ipc-types/src/reply.rs

```diff
@@ -15,6 +15,7 @@ pub struct Success {
 pub type Workspaces = Vec<Workspace>;
 
 #[derive(Deserialize, Serialize, Eq, PartialEq, Clone, Hash, Debug)]
+#[serde(from = "WorkspaceCompat")]
```

Comment:
Interesting, this is one serde attribute I'm not that familiar with. I'm having a bit of trouble finding documentation on it, can you link me to it?

My guess is it falls back on `WorkspaceCompat`if `Workspace` fails to parse?

Reply:
Oh, here: https://serde.rs/container-attrs.html#from

So it will always serialize `WorkspaceCompat` then convert into `Workspace`? I'm not sure about filling `id` out with the value of another field

Reply:
I just checked the content of the notifications from workspace subscriptions and on my system that _does_ include an id. Therefore this change is probably no good, I just assumed `id` just needed to be some unique number and didn't consider that it actually has to match other stuff.
`get_tree` seems to work on earlier versions though and its response contains the same data as `get_workspaces` with a lot of additional data. Maybe I could try implementing a shim that uses `get_tree` instead if it detects an older version, either by using `get_version` or just if `id` is missing from the response.
I'll update the pull request in a bit.